lms/djangoapps/lms_initialization/apps.py

```diff
@@ -0,0 +1,28 @@
+"""
+Initialization app for the LMS
+
+This app consists soley of a ready function in its AppConfig, and should be
+included early in the INSTALLED_APPS list.
+"""
+
+import analytics
+from django.apps import AppConfig
+from django.conf import settings
+
+
+class LMSInitializationConfig(AppConfig):
+    """
+    Application Configuration for lms_initialization.
+    """
+    name = 'lms.djangoapps.lms_initialization'
+    verbose_name = 'LMS Initialization'
+
+    def ready(self):
+        """
+        Global LMS initialization code goes here.  This runs after settings
+        have loaded, but before most other djangoapp initializations.
+        """
+
+        # Initialize Segment analytics module by setting the write_key.
+        if settings.LMS_SEGMENT_KEY:
```

Comment:
I know this is only one setting right now, but I think it would be better to isolate each fragment of startup code in to it's own function that this method would call.  I think it creates a good pattern for others to follow and will make the code easier to read/understand later.